### PR TITLE
Add `nbb.core/invoked-script` fn

### PR DESCRIPTION
**Pull request by bolivier**
Introduces `nbb.core/invoked-script` that outputs an absolute path based on the file passed to nbb, either literally or with `-m`

The rationale is to be able to use

```clj
(if (= *file* (invoked-script))
  :foo
  :bar)
```

to only run things when the script is invoked, not at a repl.

The tests verify that `*file*` and `(invoked-script)` are present and the same when used with `-m` and passing the plain script.

Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [developer documentation](https://github.com/babashka/nbb/blob/main/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/nbb/blob/main/doc/dev.md#start-with-an-issue-before-writing-code).

- [x] This PR contains a [test](https://github.com/babashka/nbb/blob/main/doc/dev.md#tests) to prevent against future regressions

- [x] I have updated the [CHANGELOG.md](https://github.com/babashka/nbb/blob/main/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

File: script/nbb_tests.clj [hunk 1]
```diff
@@ -114,6 +114,24 @@
...
+            "src/script.cljs"))))
+  (testing "calling with -m"
+    (is (= :invoked
+           (nbb {:dir "test-scripts/invoked-file-test"}
+                "-m" "core"))))
+  (testing "calling with -e"
+    (is (= :not-invoked
+           (nbb {:dir "test-scripts/invoked-file-test"}
+                "-e" "(require [nbb.core :refer [*file* invoked-file])
+(if (= *file* (invoked-file))
```
Comment (borkdude) [inline script/nbb_tests.clj:131]: This code doesn't run in nbb but in babashka, hence this part doesn't really make sense here, I think?

**Comment by borkdude** [comment]
I don't think we need to set `invoked-script` for `-m`. This also doesn't happen in babashka:

```
borkdude@m1 /tmp $ echo '(ns foo) (defn -main [& args] (prn (System/getProperty "babashka.file")))' > foo.clj
borkdude@m1 /tmp $ bb -cp . -m foo
nil
```

The reason it's not set is that `invoked-script` or `(System/getProperty "babashka.file")` is for causing side effects when the script is directly invoked.

**Comment by bolivier** [comment]
It seems like the same use cases would apply to both situations, no?

**Comment by borkdude** [comment]
No, since a `-main` function doesn't cause any side effects, unless you execute it manually, which is not the case when you invoke nbb with `-m`.

File: script/nbb_tests.clj (hunk 1 above)
Comment (bolivier) [reply script/nbb_tests.clj:131]: That should run in nbb.  It's a string passed to `nbb`

**Comment by borkdude** [comment]
Tests are failing, but the rest looks good to me

**Comment by borkdude** [comment]
Tests still fail, I think you need a quote around the vector in require. Perhaps consider running tests locally.

**Comment by borkdude** [comment]
Cool! And the last bit, can you update the changelogs? Then I'll merge

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**